### PR TITLE
feat: add netlify-plugin-next-dynamic

### PR DESCRIPTION
**Pull request by rayriffy**
Thanks for contributing the Netlify plugins directory!

**Are you adding a plugin or updating one?**

- [x] Adding a plugin
- [ ] Updating a plugin

**Have you completed the following?**

- [x] Read and followed the [plugin author guidelines](https://github.com/netlify/plugins/blob/master/docs/guidelines.md).
- [x] Included all [required fields](https://github.com/netlify/plugins/blob/master/docs/CONTRIBUTING.md#required-fields) in your entry.
- [x] Tested the plugin [locally](https://docs.netlify.com/cli/get-started/#run-builds-locally) and [on Netlify](https://docs.netlify.com/configure-builds/build-plugins/#install-a-plugin), using the plugin version stated in your entry.

**Test plan**

Basically, many people who use next as static export cannot use dynamic routes with Netlify unless they will going to **manually** write redirects. This plugin automate all of that

Full demo: https://demo-netlify-plugin-next-dynamic.netlify.app/

Source: https://github.com/Brikl/demo-netlify-plugin-next-dynamic

Successful deploy log: https://app.netlify.com/sites/demo-netlify-plugin-next-dynamic/deploys/5f1986636a6fd38c42e29609

### CODE REVIEW

**Comment by rayriffy** [comment]
If Brikl/opensource#4 is ok, then I will rollout release and update metadata and it will be ready to merge again.
